packages/core/contracts/financial-templates/common/FeePayer.sol

```diff
@@ -96,31 +96,28 @@ abstract contract FeePayer is AdministrateeInterface, Testable, Lockable {
      * This returns 0 and exit early if there is no pfc, fees were already paid during the current block, or the fee rate is 0.
      */
     function payRegularFees() public nonReentrant() returns (FixedPoint.Unsigned memory totalPaid) {
-        StoreInterface store = _getStore();
         uint256 time = getCurrentTime();
         FixedPoint.Unsigned memory collateralPool = _pfc();
-
         // Exit early if there is no collateral from which to pay fees.
         if (collateralPool.isEqual(0)) {
             // Note: set the lastPaymentTime in this case so the contract is credited for paying during periods when it
             // has no locked collateral.
             lastPaymentTime = time;
             return totalPaid;
         }
-
-        // Exit early if fees were already paid during this block.
         if (lastPaymentTime == time) {
             return totalPaid;
         }
 
         (FixedPoint.Unsigned memory regularFee, FixedPoint.Unsigned memory latePenalty) =
-            store.computeRegularFee(lastPaymentTime, time, collateralPool);
+            getOutstandingRegularFees(time);
         lastPaymentTime = time;
 
         totalPaid = regularFee.add(latePenalty);
         if (totalPaid.isEqual(0)) {
             return totalPaid;
         }
+        StoreInterface store = _getStore();
```

Comment:
this is moved lower down as if the function returns early (pays no fees) there is no need to get the store interface sooner.